bee-identity/src/identity.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Holds the `Identity` type.
+
+use std::{fmt, path::Path};
+
+use libp2p_core::{identity::PublicKey, PeerId};
+
+use crate::{migration, pem_file, Keypair as Ed25519Keypair, PublicKey as Ed25519PublicKey};
+
+/// Represents a node identity.
+#[derive(Clone, Debug)]
+pub struct Identity {
+    keypair: Ed25519Keypair,
+    is_fresh: bool,
+}
+
+impl Identity {
+    /// Creates a new identity.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Restores an [`Identity`] from a [`Keypair`](crate::ed25519::Keypair).
+    pub fn from_keypair(keypair: Ed25519Keypair) -> Self {
+        Self {
+            keypair,
+            is_fresh: false,
+        }
+    }
+
+    /// Returns the backing [`Keypair`](crate::ed25519::Keypair) of this identity.
+    pub fn keypair(&self) -> &Ed25519Keypair {
+        &self.keypair
+    }
+
+    /// Returns the [`PublicKey`](crate::ed25519::PublicKey) of this identity.
+    pub fn public_key(&self) -> Ed25519PublicKey {
+        self.keypair.public()
+    }
+
+    /// Returns the corresponding [`libp2p_core::PeerId`].
+    pub fn peer_id(&self) -> PeerId {
+        PeerId::from_public_key(&PublicKey::Ed25519(self.keypair.public()))
+    }
+
+    /// Returns whether this identity has been newly generated.
```

Comment:
what does "fresh" mean in this context? also when isn't it considered fresh anymore?  :trollface: Maybe elaborate in the comment

Reply:
The doc you marked is literally my definition of fresh in this context. Basically the same as `new`, but I got a bit bored of that term. :smile:

Reply:
Oh understand 😄 no strong opionion on this, so as you desire. i think i would just change the comment then to something like: "Returns whether this identity has been newly generated or restored file system."

Reply:
Maybe `restored_from_file` is more appropriate then?

Reply:
good idea, maybe `is_restored_from_file` ?

Reply:
changed it to `is_restored`. I think as long as we have no other source restoring from it's simpler and clear already.